packages/utils/src/cache/storages/localstorage.ts

```diff
@@ -0,0 +1,125 @@
+
+  /**
+   * Database for this class. Prefix for key of all items.
+   * to avoid key to clash with other.
+   */
+  protected dbName = '';
+
+  /**
+   * Internal cache object
+   */
+  protected cache: CacheMap | null | undefined;
+
+  /**
+   * Constructor
+   * @param name prefix key for all item
+   */
+  constructor (name: string) {
+    this.dbName = `[${StoragePrefix.DEFAULT}][${name}]`;
+    void this.getReady();
+  }
+
+  /**
+   * Prepare memory cache variable and restore all data from databases storage
+   * @returns Promise boolean
+   */
+  private async getReady (): Promise<void> {
+    try {
+      await this.restore();
+    }
+    catch (e) { // Keep it work. Even if can't connect to storage
+      this.cache = new Map();
+      // eslint-disable-next-line no-console
+      console.error(e);
+    }
+  }
+
+  /**
+   * Set a item against a key to this storage
+   * @param key Cache key
+   * @param value Data to store in cache
+   * @returns Promise void
+   */
+  public async set (key: string, value: CacheItem): Promise<void> {
+    const itemKey = getItemKey(this.dbName, key);
+    this.cache?.set(itemKey, value);
+    return Promise.resolve(localStorage.setItem(itemKey, JSON.stringify(value)));
+  }
+
+  /**
+   * Returns a item in this storage that matched by the key.
+   * @param key Cache key
+   * @returns Promise string data or `null` if nothing is cached
+   */
+  public async get (key: string): Promise<CacheItem | null> {
+    const itemKey = getItemKey(this.dbName, key);
+    return Promise.resolve(this.cache?.get(itemKey) || null);
+  }
+
+  /**
+   * Remove a item against a key to this storage
+   * @param key Cache key to remove
+   * @returns Promise void
+   */
+  public async remove (key: string): Promise<void> {
+    const itemKey = getItemKey(this.dbName, key);
+    return Promise.resolve(localStorage.removeItem(itemKey));
+  }
+
+  /**
+   * Clear all items in localStorage
+   * @returns Promise void
+   */
+  public async clear (): Promise<void> {
+    const keys = Object.keys(localStorage);
+
+    keys.filter(key => key.startsWith(this.dbName))
+      .forEach(key => {
+        localStorage.removeItem(key);
+      });
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Restore all values into memory cache
+   * @returns Promise void
+   */
+  public async restore (): Promise<void> {
+    const items = new Map<string, CacheItem>();
+    const keys = Object.keys(localStorage).filter(key => key.startsWith(this.dbName));
+
+    for (let i = 0; i < keys.length; i += 1) {
+      const item = this.retrieve(keys[i]);
+      if (item) {
+        items.set(keys[i], item);
+      }
+    }
+    this.cache = items;
+    return Promise.resolve();
+  }
+
+  /**
+   * Retrieve cache item from localStorage
+   * @param key key to retrieve value
+   * @returns data from the key
+   */
+  private retrieve (key: string): CacheItem | null {
+    try {
+      return JSON.parse(localStorage.getItem(key) || '') as CacheItem;
```

Comment:
There's no need to have catch block. You can return null right away if `localStorage.getItem(key)` is null.

Reply:
I'm not sure to remove try catch block. Because most of the way to use JSON.parse() requires try catch. And we don't know what a value of localStorage.getItem(key).